api/domain/customersDomain.js

```diff
@@ -14,53 +12,39 @@ exports.create = async (customerData) => {
 };
 
 exports.update = async (customerId, updatedCustomerData) => {
-    if (updatedCustomerData === null) {
-        throw new Erorr('Provide values to update.');
-    }
+    CustomerValidation.CustomerUpdateError(updatedCustomerData);
 
     let affectedRows = await customerRepo.updateCustomer(customerId, updatedCustomerData);
 
-    if (affectedRows === 0) {
-        throw new Error('No customer updated.');
-    }
+    CustomerValidation.CustomerAffectedRowsError(affectedRows);
     
     return affectedRows;
 };
 
 exports.delete = async (customerId) => {
-    if (isNaN(customerId)) {
-        throw new Error('Customer ID must be an integer. Given ID: ' + customerId);
-    }
+    CustomerValidation.CustomerIdError(customerId);
 
     let affectedRows = await customerRepo.deleteCustomer(customerId);
 
-    if (affectedRows === 0) {
-        throw new Error('No customer deleted.');
-    }
+    CustomerValidation.CustomerAffectedRowsError(affectedRows);
 
     return affectedRows;
 };
 
 exports.getAll = async () => {
     let fetchedRows = await customerRepo.getAll();
 
-    if (fetchedRows === 0) {
-        throw new Error('No customers found.');
-    }
+    CustomerValidation.CustomerFetchedRowsError(fetchedRows);
```

Comment:
requests that don't have any body or whatever don't need validation. They way you did it now is it's validating a result of the request, not request itself. So this isn't really what we want right now